src/humanPose/rebaScore.js

```diff
@@ -676,148 +600,96 @@ function overallRebaCalculation(skel) {
+                right: new THREE.Vector3()
+            },
+            chest: {
+                forward: new THREE.Vector3(),
+                up: new THREE.Vector3(),
+                right: new THREE.Vector3()
+            },
+            hips: {
+                forward: new THREE.Vector3(),
+                up: new THREE.Vector3(),
+                right: new THREE.Vector3()
+            }
+        }
     };
-    for (let jointId of Object.values(POSE_NET_JOINTS)) {
-        skel.joints[jointId] = humanPoseRenderer.getJointPosition(jointId);
-    }
-
-    skel.angles = getAngles(skel);
-    return skel;
+    for (let jointId of Object.values(JOINTS)) {
+        rebaData.joints[jointId] = pose.getJoint(jointId).position;
+        rebaData.scores[jointId] = 0;
+        rebaData.colors[jointId] = AnalyticsColors.undefined;
+    }
+    for (let boneId of Object.keys(JOINT_CONNECTIONS)) {
+        rebaData.boneScores[boneId] = 0;
+        rebaData.boneColors[boneId] = AnalyticsColors.undefined;
+    }
+    
+    rebaData.orientations.head.forward = rebaData.joints[JOINTS.NOSE].clone().sub(rebaData.joints[JOINTS.HEAD]).normalize();
+    rebaData.orientations.head.up = rebaData.joints[JOINTS.HEAD].clone().sub(rebaData.joints[JOINTS.NECK]).normalize();
+    rebaData.orientations.head.right = rebaData.orientations.head.forward.clone().cross(rebaData.orientations.head.up).normalize();
+    
+    rebaData.orientations.chest.up = rebaData.joints[JOINTS.NECK].clone().sub(rebaData.joints[JOINTS.CHEST]).normalize();
+    rebaData.orientations.chest.right = rebaData.joints[JOINTS.RIGHT_SHOULDER].clone().sub(rebaData.joints[JOINTS.LEFT_SHOULDER]).normalize();
+    rebaData.orientations.chest.forward = rebaData.orientations.chest.up.clone().cross(rebaData.orientations.chest.right).normalize();
+    
+    rebaData.orientations.hips.up = new THREE.Vector3(0, 1, 0); // Hips do not really have an up direction (i.e., even when sitting, the hips are always up)
+    rebaData.orientations.hips.right = rebaData.joints[JOINTS.RIGHT_HIP].clone().sub(rebaData.joints[JOINTS.LEFT_HIP]).normalize();
+    rebaData.orientations.hips.forward = rebaData.orientations.hips.up.clone().cross(rebaData.orientations.hips.right).normalize();
+
+    return rebaData;
 }
 
-function annotateHumanPoseRenderer(humanPoseRenderer) {
-    let skel = extractSkel(humanPoseRenderer);
-    calculateReba(skel);
-    humanPoseRenderer.setOverallRebaScore(overallRebaCalculation(skel));
-    for (let boneName in skel.angles) {
-        let rebaColor = skel.angles[boneName][5];
-        humanPoseRenderer.setBoneRebaColor(boneName, rebaColor);
-    }
+/**
+ * Calculates the Reba score for a given pose
+ * @param pose {Pose} The pose to calculate the score for
+ * @return {RebaData} The rebaData object
+ */
+function calculateForPose(pose) {
+    const rebaData = extractRebaData(pose);
+    calculateReba(rebaData);
+    return rebaData;
 }
 
-export {
-    annotateHumanPoseRenderer,
+export default {
```

Comment:
export default has weird mismatches in its behavior between tooling and in-browser that make it best avoided (even though it works 99% of the time)